kubernetes/resource_kubernetes_ingress.go

```diff
@@ -132,7 +139,24 @@ func resourceKubernetesIngressCreate(d *schema.ResourceData, meta interface{}) e
 	log.Printf("[INFO] Submitted new ingress: %#v", out)
 	d.SetId(buildId(out.ObjectMeta))
 
-	return resourceKubernetesIngressRead(d, meta)
+	if !d.Get("wait_for_load_balancer").(bool) {
+		return resourceKubernetesIngressRead(d, meta)
+	}
+
+	log.Printf("[INFO] Waiting for load balancer to become ready: %#v", out)
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		res, err := conn.ExtensionsV1beta1().Ingresses(metadata.Namespace).Get(metadata.Name, metav1.GetOptions{})
+		if err != nil {
```

Comment:
I suspect there can be occasions where this `err` could also be a `NotFound` caused by the API not being fast enough to register the changes from the previous `Create` call (possibly landing on another API instance)?

Although the K8s API doesn't normally behave as eventually consistent, in clusters with horizontally scaled API instances each API process maintains a local cache which can lead to brief discrepancies on the state of newly created resources depending on where the GET call lands. This is very rare, but if the Create / Get sequence is fast enough, we might still encounter it.

Reply:
Good point! I'll catch NotFound and make it a Retryable error.